### ISSUE TITLE
Interpolation of extra code with student solution

### ISSUE THREAD

**Issue by mmatos** [mumuki/mumuki-atheneum#463]
I'm making a guide for Wollok language and it happens a lot that the student needs to copy and paste a lot of code from a previous exercise because the language doesn't have open classes, which means that it's not possible to put in the extra code a class definition with the part of the logic that is required but not requested by the current exercise.

It would be great if it was possible to define a class in the extra code and then incorporate into it the new additions from the solution before appending to the final code that will be used for running the test cases.

**Comment by PalumboN** [comment]
+1

And we can aim to include open classes (and objects) in Wollok :P It would be [very useful for Wollok Game](https://github.com/uqbar-project/wollok/issues/489), too.

Another option, I'm brainstorming, can be that Mumuki _copy-pastes_ before exercise's code when current exercise hasn't submits.

**Comment by flbulgarelli** [comment]
Hi, 

definitely we need in the future a more sophisticated way of interpolating extra codes into the submission.  For example, in Haskell we usually want to add imports in the extra code, but it is not currently possible, since imports can only be in the top of the file. 

The same problem applies to Java, for example. 

In the case of wollok, interpolating methods is even more complex, since we would need to have multiple non-predefined insertion points, I can't figure out how to solve that without open clases. 

Finally, what @PalumboN asks is already reported here #455

To sum up: 

  * :+1:  to having more complex interpolations, but  this won't solve the problem described here
  * :+1: to adding open classes to Wollok. It is a really nice feature in a educative environment, that enables to build things incrementally. See https://github.com/uqbar-project/wollok/issues/609
  * :+1:  to accumulative code, but it is another issue.

**Comment by flbulgarelli** [comment]
Hi @mmatos I am now implementing a _advanced_ interpolation mechanism - initially for wollok, but which is easy to extend to other languages for such situations. 

* By default interpolation is simple: first the extra code, then the user code, finally the tests
* But you will be able to add interpolations to your tests, extra or content, that let to define where to put the remaining parts of the code, and if present, override the default interpolation. The syntax will use with `...`-comments. For example: 

```wollok
/*...extra...*/

test "...." {
  /*...content...*/
}
```

or 

 
```wollok
/*...extra...*/

test "...." {
  /*...content...*/
}
```


* Finally, you can go even finer:

**Comment by flbulgarelli** [comment]
This feature is now available to all languages. Closing
